app/merginprojectmodel.cpp

```diff
@@ -49,12 +62,28 @@ int MerginProjectModel::rowCount(const QModelIndex &parent) const {
     return mMerginProjects.count();
 }
 
-void MerginProjectModel::resetProjects()
+void MerginProjectModel::resetProjects(const ProjectList &merginProjects)
 {
     mMerginProjects.clear();
     beginResetModel();
-    mMerginProjects = mApi->projects();
+    mMerginProjects = merginProjects;
     endResetModel();
 
     emit merginProjectsChanged();
 }
+
+void MerginProjectModel::downloadProjectFinished(QString projectFolder, QString projectName)
+{
+    Q_UNUSED(projectFolder);
+    for (std::shared_ptr<MerginProject> project: mMerginProjects) {
+        if (project->name == projectName) {
+            beginResetModel();
+            project->pending = false;
+            project->status = ProjectStatus::UpToDate;
+            endResetModel();
+            emit merginProjectsChanged();
```

Comment:
Is this signal used anywhere? I can see a signal with the same name is used in MerginApi, but can't see any code that would be using this signal. If not used, best to remove it completely.